File: php/src/Behavioral/Visitor/TitleLongBlurbVisitor.php
<?

require_once 'BookInfo.php';
require_once 'GameInfo.php';
require_once 'DvdInfo.php';
require_once 'TitleBlurbVisitor.php';


class TitleLongBlurbVisitor extends TitleBlurbVisitor
{

	function visit( AbstractTitleInfo &$info )
	{
		$cname = get_class($info);
		if( $cname == 'DvdInfo' )
		{
			$this->setTitleBlurb(
				"LB-DVD: " .  $info->getTitleName() . 
				", starring " . $info->getStar() . 
				", region: " . $info->getRegion());
		}

		else if( $cname == 'BookInfo')
		{
			$this->setTitleBlurb(
				"LB-Book: " . $info->getTitleName()  .
				", Author: " .  $info->getAuthor() );
		}
		else if( $cname == 'GameInfo' )
		{
			$this->setTitleBlurb( "LB-Game: " . $info->getTitleName());
		} 
	}
}

File: csharp/src/Behavioral/Visitor/TitleLongBlurbVisitor.cs
namespace Behavioral.Visitor {


//TitleLongBlurbVisitor.java - one of two concrete Visitors

public class TitleLongBlurbVisitor : TitleBlurbVisitor {
   override public void visit(BookInfo bookInfo) {
       this.setTitleBlurb("LB-Book: " + 
                           bookInfo.getTitleName() + 
                           ", Author: " + 
                           bookInfo.getAuthor());
   }   
   
   override public void visit(DvdInfo dvdInfo) {
       this.setTitleBlurb("LB-DVD: " + 
                           dvdInfo.getTitleName() + 
                           ", starring " + 
                           dvdInfo.getStar() + 
                           ", encoding region: " + 
                           dvdInfo.getEncodingRegion());
   }   
   
   override public void visit(GameInfo gameInfo) {
       this.setTitleBlurb("LB-Game: " + 
                           gameInfo.getTitleName());
   }   
}   

}
